src/EditProvider.ts

```diff
@@ -105,14 +109,36 @@ export class EditProvider
   }
 
   public static beautifyOptions(
-    filePath: string = vscode.workspace.rootPath
+    searchStartPath: string | undefined = vscode.workspace.rootPath
   ): Promise<LanguageOptionValues> {
     try {
-      const explorer = cosmiconfig("unibeautify", {});
+      const vscodeSettings = vscode.workspace.getConfiguration("unibeautify");
+      const defaultConfigPath = vscodeSettings.defaultConfig;
+      const cosmiOptions: ExplorerOptions = {};
+      const explorer = cosmiconfig("unibeautify", cosmiOptions);
       const defaultConfig: LanguageOptionValues = {};
-      return explorer
-        .search(filePath)
-        .then(result => (result ? result.config : defaultConfig));
+
+      if (defaultConfigPath) {
+        return explorer
+          .load(defaultConfigPath)
+          .then(returnResult)
+          .catch(error => {
+            vscode.window.showErrorMessage(
+              `We could not find your configuration file: ${defaultConfigPath}.` +
+                "Please correct your path, otherwise the plugin will not work!"
+            );
+            throw error;
+          });
+      }
+
+      return explorer.search(searchStartPath).then(returnResult);
```

Comment:
See my comment above. I think `defaultConfig` should be the fallback, if `searchStartPath` (file path or workspace) does not have a `.unibeautifyrc` nearby.